File: update_version.js
const fs = require('fs');
const path = 'src/core/version.ts';
let content = fs.readFileSync(path, 'utf8');

const newVersion = "5.5.215";
const date = new Date().toISOString().split('T')[0];

content = content.replace(
    /export const SYSTEM_VERSION = ".*?";/,
    `export const SYSTEM_VERSION = "${newVersion}";`
);

const newChangelogEntry = `    {
        version: "${newVersion}",
        date: "${date}",
        changes: [
            "feat(creator): Added Zoom slider for precise waveform timing adjustments.",
            "feat(creator): Revamped Lyrics Editor with direct inline text editing.",
            "feat(creator): Added 'Merge' (chain) and 'Delete' controls for AI lyric segments.",
            "fix(creator): Resolved UI layout clipping and syntax errors in the sidebar tab system."
        ]
    },
`;

content = content.replace(
    /export const CHANGELOGS = \[\n/,
    `export const CHANGELOGS = [\n${newChangelogEntry}`
);

fs.writeFileSync(path, content);
console.log('Version updated to ' + newVersion);

File: update_version.py
import json
import datetime
import re

with open('src/core/version.ts', 'r') as f:
    content = f.read()

new_version = "5.5.271"
new_date = "2026-08-09"

changes = [
    "feat(creator): Ported Studio Timeline to Creator Studio to provide identical block dragging, ripple edit, and inline editing UX.",
    "feat(creator): Redesigned Timeline Toolbar to center editing controls and reposition track selectors, mimicking the professional Studio layout."
]

changes_str = ",\n            ".join([f'"{c}"' for c in changes])

new_changelog = f"""    {{
        version: "{new_version}",
        date: "{new_date}",
        changes: [
            {changes_str}
        ]
    }},
"""

# Replace the first instance of `export const CHANGELOGS = [` with the new changelog
content = content.replace("export const CHANGELOGS = [", f"export const CHANGELOGS = [\n{new_changelog}")

# Update SYSTEM_VERSION
content = re.sub(r'export const SYSTEM_VERSION = ".*?";', f'export const SYSTEM_VERSION = "{new_version}";', content)

# Update first line comment
content = re.sub(r'// 🛡️ v5.5.270: Creator - All-in-One Editor Redesign \(POC 1\)', f'// 🛡️ v{new_version}: Creator - All-in-One Editor Redesign (POC 3)', content)

with open('src/core/version.ts', 'w') as f:
    f.write(content)
